spec/scoreCalculatorSpec.js

```diff
@@ -0,0 +1,183 @@
+  });
+
+  describe('#calculateBaseScore', function(){
+    it('can add up the basic score of a 2, 2 roll frames', function() {
+      scoreCalculator.prepareFrameScoring(game1);
+      scoreCalculator.calculateBaseScore();
+      scoreCalculator.calculateTotalScoreFromFrames();
+      expect(scoreCalculator.totalScore).toEqual(18);
+    });
+  });
+
+  describe('#calculateSpareScore', function() {
+    it('can calculate the bonus score of spares over 4 frames', function() {
+      scoreCalculator.prepareFrameScoring(game2);
+      scoreCalculator.calculateSpareScore();
+      scoreCalculator.calculateTotalScoreFromFrames();
+      expect(scoreCalculator.totalScore).toEqual(3);
+    });
+
+    it('can calculate the bonus score of spares over 4 frames', function() {
+      scoreCalculator.prepareFrameScoring(game3);
+      scoreCalculator.calculateSpareScore();
+      scoreCalculator.calculateTotalScoreFromFrames();
+      expect(scoreCalculator.totalScore).toEqual(8);
+    });
+
+    it('can handle scoring a spare when the next turn has not happened yet', function() {
+      scoreCalculator.prepareFrameScoring(game11);
+      scoreCalculator.calculateSpareScore();
+      scoreCalculator.calculateTotalScoreFromFrames();
+      expect(scoreCalculator.totalScore).toEqual(1);
+    });
+  });
+
+    describe('#calculateTotalScore', function(){
+      it('can add up the basic and spare score of game', function() {
+        scoreCalculator.calculateTotalScore(game3);
+        expect(scoreCalculator.totalScore).toEqual(45);
+      });
+
+      it('can add up the basic, spare and strike score of game', function() {
+        scoreCalculator.calculateTotalScore(game5);
+        expect(scoreCalculator.totalScore).toEqual(65);
+      });
+    });
+
+    describe('#calculates 10 frame game correctly - no bonus scoring for 10th frame', function(){
+      it('can add up a 10th frame score with no spare or strike', function() {
+        scoreCalculator.calculateTotalScore(game6);
+        expect(scoreCalculator.totalScore).toEqual(20);
+      });
+      it('can add up a 10th frame score with a strike in the 9th frame', function() {
+        scoreCalculator.calculateTotalScore(game7);
+        expect(scoreCalculator.totalScore).toEqual(30);
+      });
+      it('can add up a 10th frame score with a stike in the 9th and 10th frame', function() {
+        scoreCalculator.calculateTotalScore(game8);
+        expect(scoreCalculator.totalScore).toEqual(53);
+      });
+      it('can add up a 10th frame score with a spare in the 10th frame', function() {
+        scoreCalculator.calculateTotalScore(game9);
+        expect(scoreCalculator.totalScore).toEqual(33);
+      });
+      it('can add up a 10th frame score with a strike in the 10th frame only', function() {
+        scoreCalculator.calculateTotalScore(game10);
+        expect(scoreCalculator.totalScore).toEqual(33);
+      });
+    });
+
+    describe('Strike scoring', function(){
+
+
```

Comment:
sounds like this method is for calculating the score when two strikes are made in a row :)